server/src/main/scala/de/frosner/broccoli/templates/SignalRefreshedTemplateSource.scala

```diff
@@ -0,0 +1,22 @@
+package de.frosner.broccoli.templates
+
+import javax.inject.Singleton
+
+import de.frosner.broccoli.models.Template
+import sun.misc.{Signal, SignalHandler}
+
+/**
+  * The template source that wraps CachedTemplateSource and refreshes the cache after receiving SIGUSR2
+  *
+  * @param source The CachedTemplateSource that will be wrapped
+  */
+@Singleton
```

Comment:
Please remove this annotation; it's not used since the source is created through a provider method which has its own `@Singleton` annotation.